### PR TITLE
Louis/indexer hash dedupe with migration

### PR DESCRIPTION
**Pull request by ldelossa**
# What
This pull adds commits necessary to reduce the copying of layer and manifest hash strings across multiple tables. 

# Why
Reduce database table size, especially in regards to the "manifest_index" table.

### CODE REVIEW

**Review by hdonnay — CHANGES_REQUESTED** [review]
The changes in `test/postgres` seem wrong, they just change the column names but don't actually change the queries from using the digests.

**Comment by ldelossa** [comment]
@hdonnay 

files in `test/postgres` fixed.

File: libindex/migrations/migration2.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	ALTER TABLE dist_scanartifact ADD FOREIGN KEY (layer_id) REFERENCES layer(id);
+	UPDATE dist_scanartifact AS ds SET layer_id = (SELECT id FROM layer WHERE hash = ds.layer_hash);
+
+	ALTER TABLE indexreport ADD COLUMN manifest_id bigint;
+	ALTER TABLE indexreport ADD FOREIGN KEY (manifest_id) REFERENCES manifest(id);
+	UPDATE indexreport AS ir SET manifest_id = (SELECT id FROM manifest WHERE hash = ir.manifest_hash);
+
+	ALTER TABLE manifest_index ADD COLUMN manifest_id bigint;
+	ALTER TABLE manifest_index ADD FOREIGN KEY (manifest_id) REFERENCES manifest(id);
+
```
Comment (Allda) [inline libindex/migrations/migration2.go:52]: There is a missing UPDATE command for the manifest_index table

Comment (ldelossa) [reply libindex/migrations/migration2.go:53]: good catch.

**Review by Allda — APPROVED** [review]
Based on the migration I did yesterday + tests of the basic use case I can confirm it works fine.

**Review by Allda — APPROVED** [review]

**Review by hdonnay — APPROVED** [review]

**Pull request merged**